pdns/recursordist/negcache.cc

```diff
@@ -86,6 +86,25 @@ bool NegCache::getRootNXTrust(const DNSName& qname, const struct timeval& now, N
   return false;
 }
 
+void NegCache::purge(const DNSName& qname, QType qtype)
+{
+  auto& mc = getMap(qname);
+  auto content = mc.lock();
+  auto& idx = content->d_map.get<NegCacheEntry>();
+
+  auto range = idx.equal_range(qname);
+  auto ni = range.first;
+
+  while (ni != range.second) {
+    // We have an entry
+    if ((ni->d_qtype == QType::ENT) || ni->d_qtype == qtype) {
+        ni = idx.erase(ni);
+    }
```

Comment:
Add `--map.d_entriesCount;` otherwise the entries count will be off.

Alternatively, we may want to extend the existing `NegCache::wipe` function to deal with `qtype == ENT`